hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/DefaultRecordStore.java

```diff
@@ -951,24 +957,43 @@ private Object putFromLoadInternal(Data key, Object value, long ttl, boolean bac
 
         Record record = getRecordOrNull(key, now, false);
         Object oldValue = null;
+        EntryEventType entryEventType = null;
         if (record == null) {
             value = mapServiceContext.interceptPut(name, null, value);
             record = createRecord(value, ttl, now);
             storage.put(key, record);
-            eventJournal.writeAddEvent(mapContainer.getEventJournalConfig(), mapContainer.getObjectNamespace(), partitionId,
-                    record.getKey(), record.getValue());
+            eventJournal.writeAddEvent(mapContainer.getEventJournalConfig(),
```

Comment:
Can you consider adding the `LOADED` event to the event journal as well as a separate PR? It would fix this issue for jet - https://github.com/hazelcast/hazelcast/issues/12224. Same in `putFromLoadInternal` and `putIfAbsent`.

Reply:
yes, that was my plan. keeping #12224 open for this.